app/models/certificate.rb

```diff
@@ -122,12 +122,13 @@ def crl_http_url
   end
 
   def aia
-    @aia ||= get_extension('authorityInfoAccess')&.
-      split(/\n/)&.
-      map { |line| line.split(/\s*-\s*/, 2) }&.
-      each_with_object(Hash.new { |hash, key| hash[key] = [] }) do |(key, value), memo|
-        memo[key] << value
-      end
+    aia_hash = parse_extension_to_hash(get_extension('authorityInfoAccess')) || {}
+    @aia ||= aia_hash
+  end
+
+  def subject_info_access
+    sia_hash = parse_extension_to_hash(get_extension('subjectInfoAccess')) || {}
+    @subject_info_access ||= sia_hash
   end
```

Comment:
these get the memoization kind of off... these do the parsing work each time and throw it away on subsequent times...

```suggestion
    @aia ||= begin
      parse_extension_to_hash(get_extension('authorityInfoAccess')) || {}
    end
  end

  def subject_info_access
    @subject_info_access ||= begin
      parse_extension_to_hash(get_extension('subjectInfoAccess')) || {}
    end
  end
```

Reply:
🤦🏼, thank you